### PR TITLE
feat(ci): tests against shell workflow debug

### PR DESCRIPTION
**Pull request by jdreSAG**
filter already tested versions in collect-shell-versions job

n/a

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/SoftwareAG/cumulocity-community-plugins/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://github.com/SoftwareAG/cumulocity-community-plugins/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: .github/workflows/test-plugins-against-cockpit.yml [hunk 1]
```diff
@@ -16,22 +17,89 @@ jobs:
...
     outputs:
-      shell_versions: ${{ steps.collect-shell-versions.outputs.shell_versions }}
+      shell_versions: ${{ steps.filter-versions.outputs.shell_versions }}
+      plugins_version: ${{ steps.extract-plugins-version.outputs.plugins_version }}
 
     steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```
Comment (reey) [inline .github/workflows/test-plugins-against-cockpit.yml:27]:
```suggestion
          fetch-depth: 1
```
`fetch-depth: 0` will fetch the complete git history, which you probably do not need in this case?

File: .github/workflows/test-plugins-against-cockpit.yml [hunk 2]
```diff
@@ -157,32 +182,107 @@ jobs:
...
 
-      - name: Upload updated JSON file to cache
-        if: ${{ env.skip_job != 'true' && github.event_name == 'schedule' }}
-        uses: actions/cache@v4
+      - name: Upload updated JSON file
+        if: ${{ github.event_name == 'schedule' }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: updated-json-${{ matrix.version_data.tag }}-${{ matrix.version_data.version }}-${{ env.PLUGINS_VERSION }}
+          path: ./cache-files/*.json
```
Comment (reey) [inline .github/workflows/test-plugins-against-cockpit.yml:214]: You could also just specify the actual file
```suggestion
          path: ./cache-files/${{ env.TAG }}-${{ env.PLUGINS_VERSION }}.json
```

File: .github/workflows/test-plugins-against-cockpit.yml [hunk 3]
```diff
@@ -157,32 +182,107 @@ jobs:
...
+
+  cache-upload:
+    permissions:
+      actions: write
+    if: ${{ github.event_name == 'schedule' && needs.collect-shell-versions.outputs.shell_versions != '[]' }}
+    needs: [collect-shell-versions, run-tests-against-shell]
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
```
Comment (reey) [inline .github/workflows/test-plugins-against-cockpit.yml:224]: Seems like there is no need to checkout the repo in this job?

Comment (jdreSAG) [reply .github/workflows/test-plugins-against-cockpit.yml:224]: it is- without it, line `gh actions-cache delete test-plugins-against-shell-e2e-results --confirm` throws an error 
https://github.com/SoftwareAG/cumulocity-community-plugins/actions/runs/10054589119/job/27789622927

File: .github/workflows/test-plugins-against-cockpit.yml (hunk 1 above)
Comment (jdreSAG) [reply .github/workflows/test-plugins-against-cockpit.yml]: probably not, will try with 1 for all occurences of checkout

Comment (jdreSAG) [reply .github/workflows/test-plugins-against-cockpit.yml]: yep, 1 is fine

**Pull request merged**